ast/Trees.cc

```diff
@@ -304,7 +304,11 @@ optional<pair<core::SymbolRef, vector<core::NameRef>>> ConstantLit::fullUnresolv
     if (this->symbol != core::Symbols::StubModule()) {
         return nullopt;
     }
-    ENFORCE(this->resolutionScopes != nullptr && !this->resolutionScopes->empty(), "loc={}", this->loc.showRaw(ctx));
+    if (this->resolutionScopes == nullptr || !this->resolutionScopes->empty()) [[unlikely]] {
```

Comment:
Should the condition in the `if` be `this->resolutionScopes == nullptr || this->resolutionScopes->empty()` (no `!` in front of `this->resolutionScopes->empty()`)?

Also, does [[unlikely]] do anything in C++17?

Reply:
good catch on the condition, thanks.

> Also, does [[unlikely]] do anything in C++17?

My understanding is that clang supports this despite it not being standardized until C++20